src/main/scala/is/hail/annotations/UnsafeRow.scala

```diff
@@ -1,73 +1,99 @@
 package is.hail.annotations
 
+import java.io.{DataInputStream, DataOutputStream}
+
+import com.esotericsoftware.kryo.io.{Input, Output}
+import com.esotericsoftware.kryo.{Kryo, KryoSerializable}
 import is.hail.expr._
 import is.hail.utils.Interval
 import is.hail.variant.{AltAllele, GenericGenotype, Locus, Variant}
 import org.apache.spark.broadcast.Broadcast
 import org.apache.spark.sql.Row
 
-class UnsafeIndexedSeqAnnotation(val region: MemoryBuffer,
-  arrayTTBc: Broadcast[TypeTree],
-  elemSize: Int, val offset: Int, elemOffset: Int,
-  val length: Int) extends IndexedSeq[Annotation] {
+class UnsafeIndexedSeqAnnotation(
+  var ttBc: Broadcast[TypeTree],
+  var region: MemoryBuffer, var aoff: Long, var elemsOffset: Long,
+  var length: Int,
+  var elemSize: Long) extends IndexedSeq[Annotation] with KryoSerializable {
   def apply(i: Int): Annotation = {
     if (i < 0 || i >= length)
       throw new IndexOutOfBoundsException(i.toString)
     assert(i >= 0 && i < length)
-    if (region.loadBit(offset + 4, i))
+    if (region.loadBit(aoff + 4, i))
       null
     else {
-      UnsafeRow.read(region, elemOffset + i * elemSize,
-        arrayTTBc.value.typ.asInstanceOf[TContainer].elementType,
-        arrayTTBc.value.subtree(0))
+      UnsafeRow.read(region, elemsOffset + i * elemSize,
+        ttBc.value.typ.asInstanceOf[TContainer].elementType,
+        ttBc.value.subtree(0))
     }
   }
+
+  override def write(kryo: Kryo, output: Output) {
+    kryo.writeObject(output, ttBc)
+    val enc = new Encoder(new DataOutputStream(output))
+    enc.writeRegionValue(ttBc.value.typ, region, aoff)
+  }
+
+  override def read(kryo: Kryo, input: Input) {
+    val cTorrentBroadcast = Class.forName("org.apache.spark.broadcast.TorrentBroadcast")
+    ttBc = kryo.readObject(input, cTorrentBroadcast).asInstanceOf[Broadcast[TypeTree]]
+    val t = ttBc.value.typ.asInstanceOf[TArray]
+
+    val dec = new Decoder(new DataInputStream(input))
+    region = MemoryBuffer()
+    aoff = dec.readRegionValue(t, region)
+
+    length = region.loadInt(aoff)
```

Comment:
I didn't change array length to Long yet.  Right now arrays are limited to max Int in length.